staging/src/k8s.io/client-go/tools/watch/retrywatcher.go

```diff
@@ -158,7 +157,7 @@ func (rw *RetryWatcher) doReceive() (bool, time.Duration) {
 				if !ok {
 					_ = rw.send(watch.Event{
 						Type:   watch.Error,
-						Object: &apierrors.NewInternalError(errors.New("retryWatcher: doesn't support resourceVersion")).ErrStatus,
+						Object: &apierrors.NewInternalError(fmt.Errorf("retryWatcher: object %s doesn't support resourceVersion", dump.Pretty(event.Object))).ErrStatus,
```

Comment:
My question still stands: does this mean that we potentially include a full dump of whatever someone is watching in an error message? Let's not do that, this can easily lead to a data leak.

How about merely including the type of the object?
```suggestion
						Object: &apierrors.NewInternalError(fmt.Errorf("retryWatcher: object type %T doesn't support resourceVersion", event.Object)).ErrStatus,
```

What was the motivation for this change? Was the old message really a problem in practice?